include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -27,7 +27,17 @@ template<typename T>
 T atomicAdd(NBL_REF_ARG(T) ptr, T value)
 {
     // FUTURE: assert the scalart type is an integer type, then assert device traits maybe? but default to traits there?
-    return spirv::atomicIAdd<T>(ptr, spv::ScopeDevice, spv::MemorySemanticsMaskNone, value);
+    if (sizeof(T) == sizeof(uint32_t))
+        return spirv::atomicIAdd<T>(ptr, spv::ScopeDevice, spv::MemorySemanticsMaskNone, value);
+    return atomic64::atomicIAdd<T>(ptr, spv::ScopeDevice, spv::MemorySemanticsMaskNone, value);
```

Comment:
why is there an `atomic64` namespace?

Use the power of `enable_if` that checks for `sizeof(T)` and `is_float_v<T>` then you can have the magic of overloads resolve automatically to the correct `spirv::` overload without having to make this function complicated

https://godbolt.nsc.devsh.eu/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:hlsl,selection:(endColumn:17,endLineNumber:19,positionColumn:2,positionLineNumber:17,selectionStartColumn:17,selectionStartLineNumber:19,startColumn:2,startLineNumber:17),source:'%23include+%22nbl/builtin/hlsl/type_traits.hlsl%22%0A%0A%0Atemplate%3Ctypename+T%3E%0Atypename+nbl::hlsl::enable_if%3Cnbl::hlsl::is_floating_point%3CT%3E::value,void%3E::type+doStuff(T+val)%0A%7B%0A++++vk::RawBufferStore%3Cfloat%3E(0,1.f)%3B%0A%7D%0Atemplate%3Ctypename+T%3E%0Atypename+nbl::hlsl::enable_if%3Cnbl::hlsl::is_integral%3CT%3E::value,void%3E::type+doStuff(T+val)%0A%7B%0A++++vk::RawBufferStore%3Cfloat%3E(0,2.f)%3B%0A%7D%0A%0A%5Bnumthreads(256,1,1)%5D%0Avoid+main()+%0A%7B%0A++++doStuff(4.5f)%3B%0A++++doStuff(14)%3B%0A%7D%0A%0A'),l:'5',n:'0',o:'HLSL+source+%231',t:'0')),k:51.44372404554398,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:nsc_release_upstream,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:hlsl,libs:!(),options:'-T+cs_6_6+-E+main+-spirv+-fspv-target-env%3Dvulkan1.3+-HV+202x',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+NSC+(Release)+(Editor+%231)',t:'0')),k:48.556275954455984,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'NSC+(Release)',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+NSC+(Release)+(Compiler+%231)',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:48.556275954455984,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4

Reply:
Done